### DISCUSSION TITLE
Adopt Paste component in MC product

### DISCUSSION THREAD

**Discussion by frankfzhao** [twilio-labs/paste#1468] [Office hours] [answered]
I am working with my PM exploring advanced segmentation for contacts in Marketing Campaign product. I have 3 different design concept we are gonna conduct usability test with customers.

Link to design: [Design that demonstrate usage on Paste design component ](https://www.figma.com/proto/j5h6mhVWU9mf6dkEtUhQ8Z/And-or-V2?page-id=234%3A0&node-id=236%3A0&viewport=-2384%2C-1611%2C0.38991665840148926&scaling=scale-down)

Link to design: [Prototype for usability test](https://www.figma.com/proto/j5h6mhVWU9mf6dkEtUhQ8Z/And-or-V2?page-id=234%3A0&node-id=236%3A0&viewport=-2384%2C-1611%2C0.38991665840148926&scaling=scale-down)

This is the first time for marketing campaign team to use Paste component and I would love some feedback from DSYS team on the following Paste design component usage:

**1. Text input**
What's the recommendation of having search in a text input?
```
    1. In the text input
    2. In the dropdown menu
```
Maximum Height of dropdown menu. 
```
What if we have over 20 options in the dropdown, when to show a scroll bar?
``` 
Maximum width of the dropdown menu
```
If option is too long, truncate or have it overflow?
```
Clear text / selection in the text input
```
Have a "x" to clear input? Should we use "suffix" or "Select icon"?
```
"suffix" or "Select icon"
```
If we have a calendar / time selector, what should we use to put the calendar / clock icon, "suffix" or "Select icon"?
```
**2. One size modal or adaptive size modal**
```
When building the logic, there will be multiple text inputs in a modal, what's the best way to deal with this?
Having a fixed width for the modal or having the modal's width to be adaptive?
```
**3. Form validation**
```
    Save not valid until all fields have bene selected?
OR
    Save always possible with error message on each input
```
**4. Visual look for the card**
```
I use shadow-low for default state and shadow for hover / select state. 
```

**Comment by kate-porter** [comment]
Hey @frankfzhao - thanks for sharing these designs, so exciting to see Paste in MC! 🙌 We'll all review the designs and your questions before office hours today. Talk soon!

**Comment by kate-porter** [comment, marked as answer]
Hey @frankfzhao - just wanted to share the office hours recording with you in case you'd like to revisit any of our conversations. [Here's the video](https://www.loom.com/share/297fac81bffa4082ad8be0a26101c4a6) (it includes timestamps for each topic), and the password is posted in the #help-design-system channel. Thanks again for coming to office hours!
